### ISSUE TITLE
Incorrect "docs" generated from the `Doxygen` in the hover

### ISSUE THREAD

**Issue by H-G-Hristov** [microsoft/vscode-cpptools#10461]
Type: <b>Bug</b>

The docs are rendered incorretly for the following header file:

```C++
#pragma once

#include <string>
#include <string_view>

#include <tl/expected.h>

namespace nbnc::register_with_server {

/**
 * @brief A simple a ADM user data property list configuration parser.
 *
 */
class IAdmInstallUserDataPlist
{
public:
    virtual ~IAdmInstallUserDataPlist() = default;

public:
    /**
     * @brief Checks if there is a plist content loaded.
     * @details The function doesn't guarantee that the loaded plist is a valid `launchd` plist.
     *
     * @return true - if there is a plist
     * @return false - if there is none.
     */
    [[nodiscard]] virtual bool IsValid() const noexcept = 0;

    /**
     * @brief Loads a `launchd` plist from file.
     * @details Use this method to read binary and text plists.
     *
     * @param filepath A full path to a plist file.
     */
    virtual tl::expected<void, std::string> LoadFromFile(std::string_view filepath) noexcept = 0;

public:
    /**
     * @brief Gets the value for `ActivationKey` key.
     * @details This optional key is used to control whether your job is to be kept continuously running or to let
     * demand and conditions control the invocation.
     *
     * @return tl::optional<std::string>
     */
    [[nodiscard]] virtual tl::optional<std::string> GetActivationKey() const noexcept = 0;

    /**
     * @brief Gets the value for `ADID` key.
     * @details  This optional key is used to control whether your job is to be kept continuously running or to let
     * demand and conditions control the invocation.
     *
     * @return tl::optional<std::string>
     */
    [[nodiscard]] virtual tl::optional<std::string> GetAdidKey() const noexcept = 0;
};

}  // namespace nbnc::register_with_server
```

![Screenshot 2023-02-03 at 16 21 47](https://user-images.githubusercontent.com/47526411/216627542-e8a798e6-df6c-419c-b282-616b6e1550ef.png)

![Screenshot 2023-02-03 at 16 21 37](https://user-images.githubusercontent.com/47526411/216627592-80f7c45b-53b0-4eb8-aa25-7dd2c0444f0c.png)

Extension version: 1.14.1
VS Code version: Code 1.75.0 (Universal) (e2816fe719a4026ffa1ee0189dc89bdfdbafb164, 2023-02-01T15:24:42.903Z)
OS version: Darwin arm64 22.2.0
Modes:
Sandboxed: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 Pro (8 x 24)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|2, 1, 1|
|Memory (System)|16.00GB (0.78GB free)|
|Process Argv|--crash-reporter-id a7eaff19-94ae-4653-b917-d9aaab19da5d|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492cf:30256860
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242cf:30382550
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
pythondataviewer:30285071
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593cf:30376535
pythonvs932:30410667
cppdebug:30492333
vsclangdc:30486549
c4g48928:30535728
dsvsc012cf:30540253
azure-dev_surveyone:30548225
vsccc:30610678
pyindex848:30577860
nodejswelcome1cf:30587006
3biah626:30602489
pyind779cf:30657577
89544117:30613380
pythonsymbol12cf:30657549
a9j8j154:30646983

```

</details>

**Comment by Colengms** [comment]
Hi @H-G-Hristov .  Would it be possible to provide an isolate repro?  Using just the code you provided, I was unable to reproduce an issue.

**Comment by H-G-Hristov** [comment]
@Colengms I am getting another strange issue in the same project:

![Screenshot 2023-02-10 at 14 22 54](https://user-images.githubusercontent.com/47526411/218091966-1c9bb5ff-5541-4bd4-88ac-10135fce6a6e.png)

It looks like a different extension is trying to render the documentation comments. This is my status bar in the same project as above:
![Screenshot 2023-02-10 at 14 26 27](https://user-images.githubusercontent.com/47526411/218092236-6cef2fc4-d28f-4aba-ba03-dc7e2008b6b8.png)

In this project I have CMake, Python, Shellscript, Swift (Xcode) code and Objective-C.


Please note the anomaly between these two screenshots:
![Screenshot 2023-02-10 at 14 28 12](https://user-images.githubusercontent.com/47526411/218092713-5c1e9581-6166-47d0-bb0d-af0d09df4e73.png)
![Screenshot 2023-02-10 at 14 29 37](https://user-images.githubusercontent.com/47526411/218092858-45baa408-9e11-42f2-a801-f51fde4af27b.png)

**Comment by H-G-Hristov** [comment]
@Colengms Please note how "Swift" appears in the namespace:

![Screenshot 2023-02-10 at 14 46 44](https://user-images.githubusercontent.com/47526411/218096193-4262bead-147a-4bb0-bb6f-365bda1fa726.png)

This could be related also to the following issue I am complaining about: https://github.com/microsoft/vscode-cpptools/issues/10413#issuecomment-1425340485

Hovering over the "Editor Language Status" displays:

> "No Package.swift"
Apple Swift version 5.7.2....

![Screenshot 2023-02-10 at 14 54 38](https://user-images.githubusercontent.com/47526411/218097377-cd51df12-a3c7-424c-9feb-63e261110fbe.png)

**Comment by Colengms** [comment]
Hi @H-G-Hristov .  I think you're correct that you are seeing results from another extension.  I'd guess that it's the Objective-C extension, but I'm not particularly knowledgeable about extensions other than the C/C++ Extension.  You could try disabling one extension at a time, to identify which is generating those results.

**Comment by H-G-Hristov** [comment]
I set   "C_Cpp.experimentalFeatures": "enabled", then I disabled the Swift extension https://marketplace.visualstudio.com/items?itemName=sswg.swift-lang
And the latter issue disappeared, which is a different the the former (original issue at the top). BTW even in Swift extension rendering the `<std::string>` string appears broken so maybe it is a string processing issue:

![Screenshot 2023-02-11 at 15 50 48](https://user-images.githubusercontent.com/47526411/218261542-495242ce-7b2b-42ac-9608-8e435f8a9c2b.png)

So basically I have two separate issues:

1. Doxygen strings are improperly rendered
2. The Swift extension takes over the C++ files resulting in various issues

Do I need to report additional issue? What would be the way forward? In my project I have C++ and Swift code I'd like to have both extensions enabled.

**Comment by sean-mcmanus** [comment]
I believe the root cause of the doxygen hover bug with is due to https://github.com/microsoft/vscode-cpptools/issues/6020 

Repro code is like:
```cpp
#include <vector>
#include <string>
/**
 * @return std::vector<std::string>
*/
std::vector<std::string> func()
{
    return std::vector<std::string>{};
}
```

**Comment by H-G-Hristov** [comment]
@sean-mcmanus Please also note on that screenshot how "Swift" appears between the filename "RegisterWithServer.hpp" and the namespace "nbnc" and if I click on it I see selectable items "Swift" and "C++" but if I select "C++" nothing changes really.
The status bar itself displays the file as C++.


> @Colengms Please note how "Swift" appears in the namespace:
> 
> ![Screenshot 2023-02-10 at 14 46 44](https://user-images.githubusercontent.com/47526411/218096193-4262bead-147a-4bb0-bb6f-365bda1fa726.png)
> 
> This could be related also to the following issue I am complaining about: [#10413 (comment)](https://github.com/microsoft/vscode-cpptools/issues/10413#issuecomment-1425340485)
> 
> Hovering over the "Editor Language Status" displays:
> 
> > "No Package.swift"
> > Apple Swift version 5.7.2....
> 
> ![Screenshot 2023-02-10 at 14 54 38](https://user-images.githubusercontent.com/47526411/218097377-cd51df12-a3c7-424c-9feb-63e261110fbe.png)

**Comment by sean-mcmanus** [comment]
I'm able to repro the bug I mentioned at https://github.com/microsoft/vscode-cpptools/issues/10461#issuecomment-1428832174 without the Swift extension.

**Comment by michelleangela** [comment]
Fix available in pre-release https://github.com/microsoft/vscode-cpptools/releases/tag/v1.17.0.

**Comment by michelleangela** [comment]
Fixed in release version https://github.com/microsoft/vscode-cpptools/releases/tag/v1.17.2.
